phpBB/includes/acp/acp_extensions.php

```diff
@@ -324,6 +351,317 @@ function main()
 		}
 	}
 
+	public function catalog_mode($id, $mode)
+	{
+		global $phpbb_container;
+
+		$action = $this->request->variable('action', 'list');
+
+		/** @var \phpbb\language\language $language */
+		$language = $phpbb_container->get('language');
+
+		/** @var \phpbb\composer\manager $composer_manager */
+		$composer_manager = $phpbb_container->get('ext.composer.manager');
+
+		/** @var \phpbb\extension\manager $extensions_manager */
+		$extensions_manager = $phpbb_container->get('ext.manager');
+
+		switch ($action)
+		{
+			case 'install':
+				$this->page_title = 'ACP_EXTENSIONS_INSTALL';
+
+				$extension = $this->request->variable('extension', '');
+
+				if (empty($extension))
+				{
+					redirect($this->u_action);
+				}
+
+				$formatter = new \phpbb\composer\io\html_output_formatter([
+					'warning' => new \Symfony\Component\Console\Formatter\OutputFormatterStyle('black', 'yellow')
+				]);
+
+				$composer_io = new \phpbb\composer\io\web_io($language, '', $phpbb_container->getParameter('extensions.composer.output'), $formatter);
+
+				try
+				{
+					$this->request->enable_super_globals();
+					$composer_manager->install((array) $extension, $composer_io);
+					$this->request->disable_super_globals();
+				}
+				catch (\phpbb\exception\runtime_exception $e)
+				{
+					$this->display_composer_exception($language, $e, $composer_io);
+					return;
+				}
+				$this->tpl_name = 'detailled_message_body';
+
+				$this->template->assign_vars(array(
+						'MESSAGE_TITLE'			=> $language->lang('ACP_EXTENSIONS_INSTALL'),
+						'MESSAGE_TEXT'			=> $language->lang('EXTENSIONS_INSTALLED') . adm_back_link($this->u_action),
+						'MESSAGE_DETAIL'		=> $composer_io->getOutput(),
+						'MESSAGE_DETAIL_LEGEND'	=> $language->lang('COMPOSER_OUTPUT'),
+						'S_USER_NOTICE'			=> true,
+					)
+				);
+
+				break;
```

Comment:
break should have the same indentation as the case. You can change phpstorm to properly do that.